spec/app/ics-029-fee-payment/README.md

```diff
@@ -0,0 +1,203 @@
+
+The sender chain will escrow 0.003 channel-7/ATOM and 0.002 IRIS. In the case that a forward relayer submits the `MsgRecvPacket` and a reverse relayer submits the `MsgAckPacket`, the forward relayer is reqarded 0.003 channel-7/ATOM and the reverse relayer is rewarded 0.001 IRIS. In the case where the packet times out, the timeout relayer receives 0.002 IRIS and 0.003 channel-7/ATOM is refunded to the original fee payer.
+
+The logic involved in collecting fees from users and then paying it out to the relevant relayers is encapsulated by a separate fee module and may vary between implementations. However, all fee modules must implement a uniform interface such that the ICS-4 handlers can correctly pay out fees to the right relayers, and so that relayers themselves can easily determine the fees they can expect for relaying a packet.
+
+### Fee Module Contract
+
+```typescript
+function PayFee(packet: Packet, forward_relayer: string, reverse_relayer: string) {
+    // pay the forward fee to the forward relayer address
+    // pay the reverse fee to the reverse relayer address
+    // refund extra tokens to original fee payer(s)
+}
+
+function PayTimeoutFee(packet: Packet, timeout_relayer: string) {
+    // pay the timeout fee to the timeout relayer address
+    // refund extra tokens to original fee payer(s)
+}
+```
+
+
+The fee module should also expose the following queries so that relayers may query their expected fee:
+
+```typescript
+// Gets the fee expected for submitting ReceivePacket msg for this packet
+function GetReceiveFee(packet) Fee
+
+// Gets the fee expected for submitting AcknowledgePacket msg for this packet
+function GetAckFee(packet) Fee
+
+// Gets the fee expected for submitting TimeoutPacket msg for this packet
+function GetTimeoutFee(packet) Fee
+```
+
+Since different chains may have different representations for fungible tokens and this information is not being sent to other chains; this ICS does not specify a particular representation for the `Fee`. Each chain may choose its own representation, it is incumbent on relayers to interpret the Fee correctly.
+
+### Connection Negotiation
+
+The chains must agree to enable the incentivization feature during the connection handshake. This can be done by adding an incentivization feature to the connection version.
+
+```{"1", ["ORDER_ORDERED", "ORDER_UNORDERED", "INCENTIVE_V1"]}```
+
+If the negotiated connection includes the incentivization feature, then the ICS-4 `WriteAcknowledgement` function must write the forward relayer address into a structured acknowledgement and the ICS-4 handlers for `AcknowledgePacket` and `TimeoutPacket` must pay fees through the ibc fee module callbacks. If the negotiated connection does not include the incentivization feature, then the ICS-4 handlers must not modify the acknowledgement provided by the application and should not call any fee callbacks even if relayer incentivization is enabled on the chain.
+
```

Comment:
I guess we don't talk about enabling relayer incentivization on chain, but on a connection level, right?

Reply:
I think we need to discuss this a bit more.

There are 2 requirements.

1. The chain must support an extra `payOnSender` field in the `IbcReceivePacket` message, so the relayer can define that.
2. We must pass that data back in the acknowledgement. This could be done in the application or in a generic Acknowledgement wrapper understood by the connection/channel. It requires both sides understand the new acknowledgement